modules/codepipeline/iam.tf

```diff
@@ -0,0 +1,68 @@
+# Set up the iam role and policy for the code pipeline.
+data "aws_iam_policy_document" "codepipeline" {
+  statement {
+    actions = [
+      "arn:aws:s3:::codepipeline*",
+      "arn:aws:s3:::elasticbeanstalk"
+    ]
+    resources = [
+      "s3:PutObject"
+    ]
+  }
+
+  statement {
+    actions = [
+      "s3:GetObject",
+      "s3:GetObjectVersion",
+      "s3:GetBucketVersioning",
+      "codecommit:CancelUploadArchive",
+      "codecommit:GetBranch",
+      "codecommit:GetCommit",
+      "codecommit:GetUploadArchiveStatus",
+      "codecommit:UploadArchive",
+      "codedeploy:CreateDeployment",
+      "codedeploy:GetApplicationRevision",
+      "codedeploy:GetDeployment",
+      "codedeploy:GetDeploymentConfig",
+      "codedeploy:RegisterApplicationRevision",
+      "elasticbeanstalk:*",
+      "ec2:*",
+      "elasticloadbalancing:*",
+      "autoscaling:*",
+      "cloudwatch:*",
+      "s3:*",
+      "sns:*",
+      "cloudformation:*",
+      "rds:*",
+      "sqs:*",
+      "ecs:*",
+      "iam:PassRole",
+      "lambda:InvokeFunction",
+      "lambda:ListFunctions"
+    ]
+    resources = [
+      "*"
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "codepipeline_assume_role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["codepipeline.amazonaws.com"]
+    }
+  }
+}
```

Comment:
open question: should we be doing the data type blocks for these IAM policies or just inserting them into the resource?

Reply:
I prefer the data type blocks just because we then don't have to worry about messing up the formatting.  Parsers aren't often the best on figuring out a bit of json or yaml in the middle of a different type of doc, so this is good for debugging.

An alternative might be to do the json directly in files and then include them.  Or if you do have preference to do inline, I don't feel that strongly about it.

Reply:
OK, I do like having the resource easier to read. I think the data block is a better option than the separate file also for readability however, the file option seems to lead us towards easier reusability so I'd be cool with which ever you prefer.